pandas/_libs/hashtable.pyx

```diff
@@ -142,7 +143,9 @@ cdef class Int64Factorizer(Factorizer):
         --------
         Factorize values with nans replaced by na_sentinel
 
-        >>> factorize(np.array([1,2,np.nan], dtype='O'), na_sentinel=20)
+        >>> fac = Int64Factorizer(3)
+        >>> arr = np.array([1,2,np.nan]).astype(np.int64)
+        >>> fac.factorize(arr, na_sentinel=20)    # doctest: +SKIP
```

Comment:
why is this skipped?

Reply:
the original example was incorrect (`object` vs. `const int64_t`).
Tried to correct but I do not understand the use-case, how can array contain NaN and be int64?

Reply:
changed the example, removed NaN